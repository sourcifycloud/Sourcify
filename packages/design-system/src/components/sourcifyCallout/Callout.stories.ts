import sourcifyCallout from './Callout.vue';
import sourcifyLink from '../sourcifyLink';
import sourcifyText from '../sourcifyText';
import type { StoryFn } from '@storybook/vue';

export default {
	title: 'Atoms/Callout',
	component: sourcifyCallout,
	argTypes: {
		theme: {
			control: {
				type: 'select',
				options: ['info', 'secondary', 'success', 'warning', 'danger', 'custom'],
			},
		},
		message: {
			control: {
				type: 'text',
			},
		},
		icon: {
			control: {
				type: 'text',
			},
		},
	},
	parameters: {
		design: {
			type: 'figma',
			url: 'https://www.figma.com/file/tPpJvbrnHbP8C496cYuwyW/Node-pinning?node-id=15%3A5777',
		},
	},
};

interface Args {
	theme: string;
	icon: string;
	default: string;
	actions: string;
	trailingContent: string;
}

const template: StoryFn<Args> = (args, { argTypes }) => ({
	props: Object.keys(argTypes),
	components: {
		sourcifyLink,
		sourcifyText,
		sourcifyCallout,
	},
	template: `
		<sourcify-callout v-bind="$props">
			${args.default}
			<template #actions v-if="actions">
				${args.actions}
			</template>
			<template #trailingContent v-if="trailingContent">
				${args.trailingContent}
			</template>
		</sourcify-callout>
	`,
});

export const defaultCallout = template.bind({});
defaultCallout.args = {
	theme: 'success',
	default: `
		This is a default callout.
	`,
};

export const customCallout = template.bind({});
customCallout.args = {
	theme: 'custom',
	icon: 'code-branch',
	default: `
		This is a custom callout.
	`,
	actions: `
		<sourcify-link size="small">
			Do something!
		</sourcify-link>
	`,
};

export const secondaryCallout = template.bind({});
secondaryCallout.args = {
	theme: 'secondary',
	icon: 'thumbtack',
	default: `
		This data is pinned.
	`,
	actions: `
		<sourcify-link theme="secondary" size="small" :bold="true" :underline="true">
			Unpin
		</sourcify-link>
	`,
	trailingContent: `
		<sourcify-link
			theme="secondary"
			size="small"
			:bold="true"
			:underline="true"
			to="https://sourcify.io"
		>
			Learn more
		</sourcify-link>
	`,
};
