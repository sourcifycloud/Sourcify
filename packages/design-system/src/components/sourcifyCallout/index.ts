import sourcifyCallout from './Callout.vue';

export default sourcifyCallout;
