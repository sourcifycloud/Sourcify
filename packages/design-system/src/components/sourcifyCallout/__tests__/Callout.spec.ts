import { render } from '@testing-library/vue';
import sourcifyCallout from '../Callout.vue';

describe('components', () => {
	describe('sourcifyCallout', () => {
		it('should render info theme correctly', () => {
			const wrapper = render(sourcifyCallout, {
				props: {
					theme: 'info',
				},
				stubs: ['sourcify-icon', 'sourcify-text'],
				slots: {
					default: '<sourcify-text size="small">This is an info callout.</sourcify-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render success theme correctly', () => {
			const wrapper = render(sourcifyCallout, {
				props: {
					theme: 'success',
				},
				stubs: ['sourcify-icon', 'sourcify-text'],
				slots: {
					default: '<sourcify-text size="small">This is a success callout.</sourcify-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render warning theme correctly', () => {
			const wrapper = render(sourcifyCallout, {
				props: {
					theme: 'warning',
				},
				stubs: ['sourcify-icon', 'sourcify-text'],
				slots: {
					default: '<sourcify-text size="small">This is a warning callout.</sourcify-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render danger theme correctly', () => {
			const wrapper = render(sourcifyCallout, {
				props: {
					theme: 'danger',
				},
				stubs: ['sourcify-icon', 'sourcify-text'],
				slots: {
					default: '<sourcify-text size="small">This is a danger callout.</sourcify-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render secondary theme correctly', () => {
			const wrapper = render(sourcifyCallout, {
				props: {
					theme: 'secondary',
				},
				stubs: ['sourcify-icon', 'sourcify-text'],
				slots: {
					default: '<sourcify-text size="small">This is a secondary callout.</sourcify-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render custom theme correctly', () => {
			const wrapper = render(sourcifyCallout, {
				props: {
					theme: 'custom',
					icon: 'code-branch',
				},
				stubs: ['sourcify-icon', 'sourcify-text'],
				slots: {
					default: '<sourcify-text size="small">This is a secondary callout.</sourcify-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render additional slots correctly', () => {
			const wrapper = render(sourcifyCallout, {
				props: {
					theme: 'custom',
					icon: 'code-branch',
				},
				stubs: ['sourcify-icon', 'sourcify-text', 'sourcify-link'],
				slots: {
					default: '<sourcify-text size="small">This is a secondary callout.</sourcify-text>',
					actions: '<sourcify-link size="small">Do something!</sourcify-link>',
					trailingContent:
						'<sourcify-link theme="secondary" size="small" :bold="true" :underline="true" to="https://sourcify.io">Learn more</sourcify-link>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
	});
});
