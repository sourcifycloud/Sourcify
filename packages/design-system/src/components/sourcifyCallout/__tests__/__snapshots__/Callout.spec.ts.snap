// Vitest Snapshot v1

exports[`components > sourcifyCallout > should render additional slots correctly 1`] = `
"<div role=\\"alert\\" class=\\"sourcify-callout callout custom\\">
  <div class=\\"message-section\\">
    <div class=\\"icon\\">
      <sourcify-icon-stub icon=\\"code-branch\\" size=\\"large\\"></sourcify-icon-stub>
    </div>
    <sourcify-text-stub size=\\"small\\" tag=\\"span\\">
      <sourcify-text-stub size=\\"small\\" tag=\\"span\\">This is a secondary callout.</sourcify-text-stub>
    </sourcify-text-stub> &nbsp; <sourcify-link-stub size=\\"small\\">Do something!</sourcify-link-stub>
  </div>
  <sourcify-link-stub theme=\\"secondary\\" size=\\"small\\" bold=\\"true\\" underline=\\"true\\" to=\\"https://sourcify.io\\">Learn more</sourcify-link-stub>
</div>"
`;

exports[`components > sourcifyCallout > should render custom theme correctly 1`] = `
"<div role=\\"alert\\" class=\\"sourcify-callout callout custom\\">
  <div class=\\"message-section\\">
    <div class=\\"icon\\">
      <sourcify-icon-stub icon=\\"code-branch\\" size=\\"large\\"></sourcify-icon-stub>
    </div>
    <sourcify-text-stub size=\\"small\\" tag=\\"span\\">
      <sourcify-text-stub size=\\"small\\" tag=\\"span\\">This is a secondary callout.</sourcify-text-stub>
    </sourcify-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > sourcifyCallout > should render danger theme correctly 1`] = `
"<div role=\\"alert\\" class=\\"sourcify-callout callout danger\\">
  <div class=\\"message-section\\">
    <div class=\\"icon\\">
      <sourcify-icon-stub icon=\\"times-circle\\" size=\\"large\\"></sourcify-icon-stub>
    </div>
    <sourcify-text-stub size=\\"small\\" tag=\\"span\\">
      <sourcify-text-stub size=\\"small\\" tag=\\"span\\">This is a danger callout.</sourcify-text-stub>
    </sourcify-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > sourcifyCallout > should render info theme correctly 1`] = `
"<div role=\\"alert\\" class=\\"sourcify-callout callout info\\">
  <div class=\\"message-section\\">
    <div class=\\"icon\\">
      <sourcify-icon-stub icon=\\"info-circle\\" size=\\"large\\"></sourcify-icon-stub>
    </div>
    <sourcify-text-stub size=\\"small\\" tag=\\"span\\">
      <sourcify-text-stub size=\\"small\\" tag=\\"span\\">This is an info callout.</sourcify-text-stub>
    </sourcify-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > sourcifyCallout > should render secondary theme correctly 1`] = `
"<div role=\\"alert\\" class=\\"sourcify-callout callout secondary\\">
  <div class=\\"message-section\\">
    <div class=\\"icon\\">
      <sourcify-icon-stub icon=\\"info-circle\\" size=\\"medium\\"></sourcify-icon-stub>
    </div>
    <sourcify-text-stub size=\\"small\\" tag=\\"span\\">
      <sourcify-text-stub size=\\"small\\" tag=\\"span\\">This is a secondary callout.</sourcify-text-stub>
    </sourcify-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > sourcifyCallout > should render success theme correctly 1`] = `
"<div role=\\"alert\\" class=\\"sourcify-callout callout success\\">
  <div class=\\"message-section\\">
    <div class=\\"icon\\">
      <sourcify-icon-stub icon=\\"check-circle\\" size=\\"large\\"></sourcify-icon-stub>
    </div>
    <sourcify-text-stub size=\\"small\\" tag=\\"span\\">
      <sourcify-text-stub size=\\"small\\" tag=\\"span\\">This is a success callout.</sourcify-text-stub>
    </sourcify-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > sourcifyCallout > should render warning theme correctly 1`] = `
"<div role=\\"alert\\" class=\\"sourcify-callout callout warning\\">
  <div class=\\"message-section\\">
    <div class=\\"icon\\">
      <sourcify-icon-stub icon=\\"exclamation-triangle\\" size=\\"large\\"></sourcify-icon-stub>
    </div>
    <sourcify-text-stub size=\\"small\\" tag=\\"span\\">
      <sourcify-text-stub size=\\"small\\" tag=\\"span\\">This is a warning callout.</sourcify-text-stub>
    </sourcify-text-stub> &nbsp;
  </div>
</div>"
`;
